Imports in the mjs do not work out of the box with node ESM

### Describe the bug

Generated `dist/index.mjs` :
```
import { Migration } from "./migration";
import { PgType } from "./types";
import { isPgLiteral, PgLiteral } from "./utils";
import runner from "./runner";
```

To work out of the box with ESM, it should be : 
```
import { Migration } from "./migration.js";
import { PgType } from "./types.js";
import { isPgLiteral, PgLiteral } from "./utils.js";
import runner from "./runner.js";
```

### Steps to reproduce

Create a `myMigrate.mjs`: 
```
import nodePgMigrate, {RunnerOption} from 'node-pg-migrate';

await nodePgMigrate.default({...})
```

### Logs

```
node:internal/modules/esm/resolve:264
    throw new ERR_MODULE_NOT_FOUND(
          ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/remidewitte/Work/francefrais/node_modules/node-pg-migrate/dist/migration' imported from /Users/remidewitte/Work/francefrais/node_modules/node-pg-migrate/dist/index.mjs
    at finalizeResolution (node:internal/modules/esm/resolve:264:11)
    at moduleResolve (node:internal/modules/esm/resolve:924:10)
    at defaultResolve (node:internal/modules/esm/resolve:1148:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:390:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:359:25)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:234:38)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:87:39)
    at link (node:internal/modules/esm/module_job:86:36) {
  code: 'ERR_MODULE_NOT_FOUND',
  url: 'file:///Users/remidewitte/Work/francefrais/node_modules/node-pg-migrate/dist/migration'
}
```

### System Info

```shell
System:
    OS: macOS 13.1
    CPU: (8) arm64 Apple M1 Pro
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 21.7.2 - /opt/homebrew/bin/node
    npm: 10.5.0 - /opt/homebrew/bin/npm
```


### Used Module System

esm

--- a/.gitignore
+++ b/.gitignore
@@ -82,6 +82,7 @@ versions.json
 /migrations
 /bin/*.js
 /bin/*.mjs
+/bin/esm
 
 TAGS
 REVISION

--- a/bin/node-pg-migrate.ts
+++ b/bin/node-pg-migrate.ts
@@ -3,6 +3,8 @@
 import type { DotenvConfigOptions } from 'dotenv';
 // Import as node-pg-migrate, so tsup does not self-reference as '../dist'
 // otherwise this could not be imported by esm
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment, @typescript-eslint/prefer-ts-expect-error
+// @ts-ignore: when a clean was made, the types are not present in the first run
 import { default as migrationRunner, Migration } from 'node-pg-migrate';
 import { readFileSync } from 'node:fs';
 import { resolve } from 'node:path';
@@ -464,10 +466,16 @@ if (action === 'create') {
           ignorePattern: IGNORE_PATTERN,
         }),
   })
-    .then((migrationPath) => {
-      console.log(format('Created migration -- %s', migrationPath));
-      process.exit(0);
-    })
+    .then(
+      (
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment, @typescript-eslint/prefer-ts-expect-error
+        // @ts-ignore: when a clean was made, the types are not present in the first run
+        migrationPath
+      ) => {
+        console.log(format('Created migration -- %s', migrationPath));
+        process.exit(0);
+      }
+    )
     .catch((error: unknown) => {
       console.error(error);
       process.exit(1);

--- a/bin/node-pg-migrate.ts
+++ b/bin/node-pg-migrate.ts
@@ -1,13 +1,16 @@
 #!/usr/bin/env node
 
 import type { DotenvConfigOptions } from 'dotenv';
+// Import as node-pg-migrate, so tsup does not self-reference as '../dist'
+// otherwise this could not be imported by esm
+import { default as migrationRunner, Migration } from 'node-pg-migrate';
 import { readFileSync } from 'node:fs';
 import { resolve } from 'node:path';
 import { format } from 'node:util';
 import type { ClientConfig } from 'pg';
-import ConnectionParameters from 'pg/lib/connection-parameters';
+// This needs to be imported with .js extension, otherwise it will fail in esm
+import ConnectionParameters from 'pg/lib/connection-parameters.js';
 import yargs from 'yargs/yargs';
-import { default as migrationRunner, Migration } from '../dist';
 import type { RunnerOption } from '../src';
 import type { FilenameFormat } from '../src/migration';
 

--- a/package.json
+++ b/package.json
@@ -4,7 +4,7 @@
   "description": "PostgreSQL database migration management tool for node.js",
   "scripts": {
     "clean": "rimraf .eslintcache dist pnpm-lock.yaml node_modules",
-    "build:bin": "tsup-node ./bin/node-pg-migrate.ts --clean false --out-dir ./bin",
+    "build:bin": "tsup-node --config tsup-bin.config.ts",
     "build:clean": "rimraf dist",
     "build:code": "tsup-node",
     "build:types": "tsc --project tsconfig.build.json",
@@ -28,31 +28,29 @@
   },
   "type": "commonjs",
   "main": "dist/index.js",
-  "module": "dist/index.mjs",
+  "module": "dist/esm/index.mjs",
   "types": "dist/index.d.ts",
   "exports": {
     "./dist/*": {
       "types": "./dist/*.d.ts",
       "require": "./dist/*.js",
-      "import": "./dist/*.mjs",
+      "import": "./dist/esm/*.mjs",
       "default": "./dist/*.js"
     },
-    "./bin/node-pg-migrate": {
-      "require": "./bin/node-pg-migrate.js",
-      "import": "./bin/node-pg-migrate.mjs",
-      "default": "./bin/node-pg-migrate.js"
+    "./bin/*": {
+      "require": "./bin/*.js",
+      "import": "./bin/*.mjs",
+      "default": "./bin/*.js"
     },
     ".": {
       "types": "./dist/index.d.ts",
       "require": "./dist/index.js",
-      "import": "./dist/index.mjs",
+      "import": "./dist/esm/index.mjs",
       "default": "./dist/index.js"
     },
     "./*": {
       "types": "./dist/*.d.ts",
-      "require": "./dist/*.js",
-      "import": "./dist/*.mjs",
-      "default": "./dist/*.js"
+      "require": "./dist/*.js"
     },
     "./package.json": "./package.json"
   },
@@ -122,6 +120,7 @@
     "eslint-plugin-prettier": "5.1.3",
     "eslint-plugin-unicorn": "52.0.0",
     "json5": "2.2.3",
+    "node-pg-migrate": "file:.",
     "npm-run-all2": "6.1.2",
     "pg": "8.11.5",
     "prettier": "3.2.5",

--- a/src/db.ts
+++ b/src/db.ts
@@ -3,6 +3,7 @@
  */
 import { inspect } from 'node:util';
 import type {
+  Client,
   ClientBase,
   ClientConfig,
   QueryArrayConfig,
@@ -10,7 +11,8 @@ import type {
   QueryConfig,
   QueryResult,
 } from 'pg';
-import { Client } from 'pg';
+// This needs to be imported as `*`, otherwise it will fail in esm
+import * as pg from 'pg';
 import type { DB, Logger } from './types';
 
 /* eslint-disable @typescript-eslint/no-explicit-any */
@@ -54,7 +56,7 @@ function db(
 
   const client: Client = isExternalClient
     ? (connection as Client)
-    : new Client(connection as string | ClientConfig);
+    : new pg.Client(connection as string | ClientConfig);
 
   const beforeCloseListeners: any[] = [];
 

--- a/tsup-bin.config.ts
+++ b/tsup-bin.config.ts
@@ -1,0 +1,13 @@
+import { defineConfig } from 'tsup';
+
+export default defineConfig({
+  entry: ['bin/node-pg-migrate.ts'],
+  outDir: 'bin',
+  clean: false,
+  format: ['esm', 'cjs'],
+  target: ['es2020', 'node16'],
+  dts: false,
+  minify: false,
+  sourcemap: false,
+  bundle: false,
+});

--- a/tsup.config.ts
+++ b/tsup.config.ts
@@ -1,13 +1,29 @@
 import { defineConfig } from 'tsup';
 
-export default defineConfig({
-  entry: ['src/**/*.ts'],
-  outDir: 'dist',
-  clean: true,
-  format: ['esm', 'cjs'],
-  target: ['es2020', 'node16'],
-  dts: false,
-  minify: false,
-  splitting: true,
-  bundle: false,
-});
+export default defineConfig([
+  // src esm
+  {
+    entry: ['src/index.ts'],
+    outDir: 'dist/esm',
+    clean: true,
+    format: 'esm',
+    target: ['es2020', 'node16'],
+    dts: false,
+    minify: false,
+    sourcemap: false,
+    bundle: true,
+    outExtension: () => ({ js: '.mjs' }),
+  },
+  // src cjs
+  {
+    entry: ['src/**/*.ts'],
+    outDir: 'dist',
+    clean: false,
+    format: 'cjs',
+    target: ['es2020', 'node16'],
+    dts: false,
+    minify: false,
+    sourcemap: false,
+    bundle: false,
+  },
+]);

